importlib_metadata/__init__.py

```diff
@@ -440,7 +440,7 @@ def at(path: str | os.PathLike[str]) -> Distribution:
         :param path: a string or path-like object
         :return: a concrete Distribution instance for the path
         """
-        return PathDistribution(pathlib.Path(path))
+        return PathDistribution(cast(SimplePath, pathlib.Path(path)))
```

Comment:
I notice that the commit message says this change is unrelated to the PR, but it does seem to be related, as tests are passing in main but not against this PR rebased to main. I'd like to understand better why this change is necessary (and that it's correct).

Reply:
Reverting 4a3782d11895c22fd443c67175ab7352db2a769b, I see the following error from mypy:

```
443: error: Argument 1 to "PathDistribution" has incompatible type "Path"; expected "SimplePath"  [arg-type]
443: note: Following member(s) of "Path" have conflicts:
443: note:     Expected:
443: note:         def __truediv__(self, str | PathLike[str], /) -> SimplePath
443: note:     Got:
443: note:         def __truediv__(self, str | PathLike[str], /) -> Path
443: note:     Expected:
443: note:         def joinpath(self, other: str | PathLike[str]) -> SimplePath
443: note:     Got:
443: note:         def joinpath(self, *other: str | PathLike[str]) -> Path
443: note:     <2 more conflict(s) not shown>
```

It seems somehow `Path` isn't considered a `SimplePath`. Oh, perhaps it's the introduction of the `def resolve()` to the `SimplePath` protocol.

Reply:
Yes, that was the case. Removing the `def resolve` from the `SimplePath` spec resolves the issue.